core/src/translate/ast_literal.rs

```diff
@@ -23,8 +23,12 @@ pub fn translate_ast_literal(sql_value: &SqlValue) -> Result<AstLiteral> {
             ..
         } => AstLiteral::Interval {
             value: value.to_owned(),
-            leading_field: leading_field.as_ref().map(translate_datetime_field),
-            last_field: last_field.as_ref().map(translate_datetime_field),
+            leading_field: leading_field
+                .as_ref()
+                .map(|x| translate_datetime_field(x).unwrap()),
+            last_field: last_field
+                .as_ref()
+                .map(|x| translate_datetime_field(x).unwrap()),
```

Comment:
```rust
leading_field: leading_field      
    .as_ref()                     
    .map(translate_datetime_field)
    .transpose()?,                
last_field: last_field            
    .as_ref()                     
    .map(translate_datetime_field)
    .transpose()?,                
```

I checked in the last review but forgot to mention it.
This is the last one, everything else looks great. Thanks! :)